test/TritonGPU/dot-operands.mlir

```diff
@@ -110,9 +110,10 @@ module attributes {"ttg.target" = "cuda:90", "ttg.num-ctas" = 1 : i32, "ttg.num-
 #tmem_scales = #ttng.tensor_memory_scales_encoding<>
 module attributes {"ttg.num-ctas" = 1 : i32, "ttg.num-warps" = 4 : i32, ttg.target = "cuda:100", "ttg.threads-per-warp" = 32 : i32} {
   // CHECK-LABEL: @scales_in_shmem
+  // FIXME
   // CHECK: %[[A_LA:.*]] = ttg.local_alloc
   // CHECK: %[[B_LA:.*]] = ttg.local_alloc
-  // CHECK: ttng.tc_gen5_mma_scaled {{.*}}, %[[A_LA]], %[[B_LA]],
+  // CHECK-NOT: ttng.tc_gen5_mma_scaled {{.*}}, %[[A_LA]], %[[B_LA]],
```

Comment:
This is broken because the unswizzled TMA layout is no longer innermost contiguous here:
https://github.com/triton-lang/triton/blob/85571488811ec8fea77acb104125b3d3d0bc5701/lib/Dialect/TritonGPU/Transforms/OptimizeDotOperands.cpp#L315

Reply:
Regarding that if, it's unclear to me whether that's the weakest preconditoin we can have. Happy to discuss it on Monday see if we can tighten it so that we can still generate the dot_scaled op

Reply:
> This is broken because the unswizzled TMA layout is no longer innermost contiguous

Why is that the case? I didn't expect it to change for existing shapes that don't have outter size > 256

Reply:
The shape here is 2x512 so it needs to be split into 2x256 tma messages.

Reply:
I resolved this by reverting to `swizzle_shared` in the lit test, which I think is still relevant for the `AsyncCopy` case. I also realized the TMEMCopy lowering would actually be broken by this PR since it assumes a completely contiguous layout, so I've updated it to use linear layouts to calculate the smem addresses.

Reply:
but the TMA cases are still going to be broken? I think this is something we should also test as we want it working?

Reply:
There is this lit test that checks the 5D case works, it's just the 2D case we can't support just now.
https://github.com/triton-lang/triton/blob/71b7af17ad1e6d404bb645fff9e69fe3ab415201/test/TritonGPU/dot-operands.mlir#L186

Reply:
ok got it, as long as we don't regress existing kernels we should be good then